src/Templating/MutableTemplateRegistryInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Twig\Templating;
+
+/**
+ * @author Timo Bakx <timobakx@gmail.com>
+ */
+interface MutableTemplateRegistryInterface extends TemplateRegistryInterface
+{
+    /**
+     * NEXT_MAJOR: remove this method declaration with docblock and uncomment code below.
+     *
+     * @param array<string, string> $templates 'name' => 'file_path.html.twig'
+     */
+    public function setTemplates(array $templates);
+
+    ///**
+    // * @param array<string, string> $templates 'name' => 'file_path.html.twig'
+    // */
+    //public function setTemplates(array $templates): void;
```

Comment:
Same here

Reply:
I can't becouse it have to be the same as in AdminBundle:

https://github.com/sonata-project/SonataAdminBundle/pull/6755

Reply:
The solution would be to deprecate the ones in AdminBundle in the 4.x version then